src/lib/Microsoft.Health.Events/EventConsumers/StreamingEventConsumerService.cs

```diff
@@ -0,0 +1,141 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.Health.Events.Common;
+using Microsoft.Health.Events.EventCheckpointing;
+using Microsoft.Health.Events.EventConsumers.Service;
+using Microsoft.Health.Events.Model;
+using Microsoft.Health.Events.Telemetry;
+using Microsoft.Health.Logging.Telemetry;
+
+namespace Microsoft.Health.Events.EventConsumers
+{
+    public abstract class StreamingEventConsumerService : IEventConsumerService
+    {
+        protected const string ThresholdWaitReached = nameof(ThresholdWaitReached);
+        protected const string ThresholdCountReached = nameof(ThresholdCountReached);
+        protected const int PartitionCheckpointThreshold = 100;

```

Comment:
I'd add this into the `EventBatchingOptions`, with a default of 100. That allows us to experiment with this value by changing an environment variable.